### ISSUE TITLE
Reading a PDF and saving it results in an empty PDF

### ISSUE THREAD

**Issue by moritzgloeckl** [gdelugre/origami#33]
Hi,

I'm trying to read a PDF created by https://github.com/prawnpdf/prawn, it seems fine while reading it (no errors), however after saving it, it's just a blank page...

When comparing it an HexEditor (before and after), the major difference (besides Origami removing blanks and updating the xrefs) is this:

```
8 0 obj
<<
	/Parent 3 0 R
	/Resources <<
		/Font <<
			/F1 <<
				/Type /Font
				/Subtype /Type1
				/BaseFont /Helvetica
			>>
		>>
	>>
```

So it seems Origami is embedding a new Font? I can see the font being defined earlier...

Any ideas? Thanks!

Edit: `pdf.pages.size` results to 0 so I guess the error happens when reading it

**Comment by moritzgloeckl** [comment]
Okay got it working, the difference is that in my original PDF it says `/Kids [5 0 R]` and Origami changes that to `/Kids [8 0 R]` I'm not sure why it does that but changing it back fixes my PDF...

**Comment by moritzgloeckl** [comment]
@gdelugre Hey I think that this is a much bigger bug than previously thought, because it happens with every PDF.

I can't open any PDF because the /Kids array is wrong...

**Comment by gdelugre** [comment]
Fixed at 9282cae738d7e33ef06dc05b976b4a2e2571440d

**Comment by cdodhia** [comment]
> Hi,
> 
> I'm trying to read a PDF created by https://github.com/prawnpdf/prawn, it seems fine while reading it (no errors), however after saving it, it's just a blank page...
> 
> When comparing it an HexEditor (before and after), the major difference (besides Origami removing blanks and updating the xrefs) is this:
> 
> ```
> 8 0 obj
> <<
> 	/Parent 3 0 R
> 	/Resources <<
> 		/Font <<
> 			/F1 <<
> 				/Type /Font
> 				/Subtype /Type1
> 				/BaseFont /Helvetica
> 			>>
> 		>>
> 	>>
> ```
> 
> So it seems Origami is embedding a new Font? I can see the font being defined earlier...
> 
> Any ideas? Thanks!
> 
> Edit: `pdf.pages.size` results to 0 so I guess the error happens when reading it

Hey @gdelugre  Can you share the script of reading from Prawn Document?
